Automattic/woocommerce-payments
# Fix fatal error when a subscription renews automatically

Fixes #3892

#### Changes proposed in this Pull Request

* Add a check to make sure there's a session before trying to unset a value in it.
* As far as I can tell, no other methods that modify the session are called without the possibility of a session existing.

#### Testing instructions

* Install WooCommerce Subscriptions. This can be obtained from the `all-plugins` repo.
* Enable UPE checkout if not already enabled.
* Create a subscription product that renews daily, make sure it has a price (like 10).
* Add subscription to your cart and check out.
* Modify the `woocommerce-subscriptions/vendor/woocommerce/subscriptions-core/assets/js/admin/meta-boxes-subscription.js` file by commenting out lines 206-232. This allows you to set the subscription to renew in less than an hour.
* Go to Subscriptions, and then into your new subscription.
* Set the _Next payment_ date for the current day and about 2 minutes into the future.
* Log out of WordPress.
* Set a breakpoint in the [`remove_upe_payment_intent_from_session` method](https://github.com/Automattic/woocommerce-payments/pull/3893/files#diff-b541c0a287f40f82a1d0e43c52c7c7a2b67ae233f261655cbb437726fec83feeR1133) and wait for the renewal to process.
* When the renewal processes, it should find no session and should skip over calling `__unset`. 

-------------------


**Post merge**
